### PR TITLE
Improve CI infrastructure and setup

### PR DESCRIPTION
**Pull request by drapergeek**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,13 +2,10 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/elixir:1.6.1-browsers
+      - image: circleci/elixir:1.6.4-browsers
```
Comment (dbernheisel) [inline .circleci/config.yml:5]: This doesn't match what's in the project's `.tool-version`. Would you bump that too? Or bump the version separately in another PR?

File: .circleci/config.yml [hunk 2]
```diff
@@ -27,7 +24,7 @@ jobs:
           - npm-cache-{{ arch }}-{{ .Branch }}-{{ checksum "package-lock.json" }}
           - npm-cache-{{ arch }}-{{ .Branch }}
           - npm-cache-{{ arch }}
-      - run: bin/setup
+      - run: mix local.hex --force && mix local.rebar --force && mix deps.get
```
Comment (dbernheisel) [inline .circleci/config.yml:27]: Why not `bin/setup`?

**Comment by dbernheisel** [comment]
We're already using Circle 2.0. Is there a better PR title? I'm not actually clear what you're trying to do in this one.

File: .circleci/config.yml (hunk 1 above)
Comment (drapergeek) [reply .circleci/config.yml:5]: May as well bump it here, not a big bump anyway.

File: .circleci/config.yml (hunk 2 above)
Comment (drapergeek) [reply .circleci/config.yml:27]: There is way more setup needed on a dev machine than on CI. I think at this point I'm going to remove the CI stuff in the `bin/setup` and pus another commit.

**Comment by drapergeek** [comment]
> We're already using Circle 2.0. Is there a better PR title? I'm not actually clear what you're trying to do in this one.

Changed title

**Pull request merged**